core/src/main/kotlin/org/lflang/generator/ts/TSGenerator.kt

```diff
@@ -123,15 +123,15 @@ class TSGenerator(
 
         val codeMaps = HashMap<Path, CodeMap>()
         generateCode(codeMaps, resource.model.preambles)
-        if (targetConfig.dockerOptions != null) {
+        if (targetConfig.dockerOptions.get().enabled) {
             val dockerData = TSDockerGenerator(context).generateDockerData();
             dockerData.writeDockerFile()
             DockerComposeGenerator(context).writeDockerComposeFile(listOf(dockerData))
         }
         // For small programs, everything up until this point is virtually instantaneous. This is the point where cancellation,
         // progress reporting, and IDE responsiveness become real considerations.
 
-        if (context.mode != LFGeneratorContext.Mode.LSP_MEDIUM && targetConfig.noCompile) {
+        if (context.mode != LFGeneratorContext.Mode.LSP_MEDIUM && targetConfig.noCompile.get()) {
```

Comment:
`NoCompileProperty` does not return TS in `supportedTargets` method.

Reply:
Nice catch!

Reply:
That brings me to another idea: Maybe we should at an assertion to the `get()` and `toString` (maybe also others) methods of the Property class and check that the property is enabled for the current target. Then we would be able to catch those bugs more easily during testing.

Reply:
That would require making the property aware of the current target (e.g., by passing a reference to the associated `TargetConfig` into the constructor). I'm hesitant to do this because those methods are not final but meant to be overridden, meaning the assertion could easily be bypassed...